### PR TITLE
Adjust docs for password protected keystore

### PR DESCRIPTION
**Pull request by jkakavas**
This commit adds relevant parts in the elasticsearch-keystore
sub-commands reference docs and in the reload secure settings API
doc.

### CODE REVIEW

File: docs/reference/cluster/nodes-reload-secure-settings.asciidoc [hunk 1]
```diff
@@ -1,9 +1,9 @@
 [[cluster-nodes-reload-secure-settings]]
 == Nodes Reload Secure Settings
 
-The cluster nodes reload secure settings API is used to re-read the
-local node's encrypted keystore. Specifically, it will prompt the keystore
-decryption and reading across the cluster. The keystore's plain content is
+The cluster nodes reload secure settings API is used to re-read the local node's encrypted keystore.
+Specifically, it will prompt the keystore decryption and reading across the cluster.
+The keystore's decrypted content is
 used to reinitialize all compatible plugins. A compatible plugin can be
```
Comment (rjernst) [inline docs/reference/cluster/nodes-reload-secure-settings.asciidoc]: I think this existing wording is deceptive. Plugin's can't be reinitialized. Rather, the consumers of the keystore settings are notified of the updated value.

File: docs/reference/cluster/nodes-reload-secure-settings.asciidoc [hunk 2]
```diff
@@ -22,9 +22,37 @@ The first command reloads the keystore on each node. The seconds allows
 to selectively target `nodeId1` and `nodeId2`. The node selection options are
 detailed <<cluster-nodes,here>>.
 
-Note: It is an error if secure settings are inconsistent across the cluster
-nodes, yet this consistency is not enforced whatsoever. Hence, reloading specific
-nodes is not standard. It is only justifiable when retrying failed reload operations.
+==== Reload Password Protected Secure Settings
+
+When the elasticsearch keystore is password protected and not simply obfuscated, the password for the keystore needs to be provided in the request to reload the secure settings.
+Reloading the settings for the whole cluster assumes that all nodes keystores are protected with the same password and is only allowed when
```
Comment (rjernst) [inline docs/reference/cluster/nodes-reload-secure-settings.asciidoc]: nit: `nodes` -> `nodes'` (plural possessive)

File: docs/reference/cluster/nodes-reload-secure-settings.asciidoc [hunk 3]
```diff
@@ -1,9 +1,9 @@
 [[cluster-nodes-reload-secure-settings]]
 == Nodes Reload Secure Settings
 
-The cluster nodes reload secure settings API is used to re-read the
-local node's encrypted keystore. Specifically, it will prompt the keystore
-decryption and reading across the cluster. The keystore's plain content is
+The cluster nodes reload secure settings API is used to re-read the local node's encrypted keystore.
```
Comment (rjernst) [inline docs/reference/cluster/nodes-reload-secure-settings.asciidoc]: The `local node's encrypted keystore` implies to me the api must be run on every node, when in fact this is a cluster level API, correct? Maybe reword to `...is used to re-load the keystore on each node`? I'm not sure `encrypted` should be there as a qualifier since in the no-passphrase case it is only obfuscated.

File: docs/reference/cluster/nodes-reload-secure-settings.asciidoc [hunk 4]
```diff
@@ -22,9 +22,37 @@ The first command reloads the keystore on each node. The seconds allows
...
+[source,js]
+--------------------------------------------------
+POST _nodes/reload_secure_settings
+{
+  "reload_secure_settings": "s3cr3t" <1>
+}
+--------------------------------------------------
+// NOTCONSOLE
+
+<1> The password that the elasticsearch keystore is encrypted with.
```
Comment (rjernst) [inline docs/reference/cluster/nodes-reload-secure-settings.asciidoc]: I woudl add a qualfiier that this password must be the same on all nodes.

File: docs/reference/cluster/nodes-reload-secure-settings.asciidoc [hunk 5]
```diff
@@ -22,9 +22,37 @@ The first command reloads the keystore on each node. The seconds allows
...
+[source,js]
+--------------------------------------------------
+POST _nodes/_local/reload_secure_settings
+{
+  "reload_secure_settings": "s3cr3t" <1>
+}
+--------------------------------------------------
+// NOTCONSOLE
+
+<1> The password that the elasticsearch keystore is encrypted with.
```
Comment (rjernst) [inline docs/reference/cluster/nodes-reload-secure-settings.asciidoc]: encrypted with _on the local node_

File: docs/reference/cluster/nodes-reload-secure-settings.asciidoc [hunk 6]
```diff
@@ -22,9 +22,37 @@ The first command reloads the keystore on each node. The seconds allows
...
+
+Note:It is an error if secure settings are inconsistent across the cluster nodes, yet this consistency is not enforced whatsoever.Hence, reloading specific nodes is not standard.It is only justifiable when retrying failed reload operations.
```
Comment (rjernst) [inline docs/reference/cluster/nodes-reload-secure-settings.asciidoc]: Should this be a real asciidoc "Note"? Also, seem to have lost some spacing between sentences in moving down here.

File: docs/reference/setup/secure-settings.asciidoc [hunk 7]
```diff
@@ -32,16 +28,33 @@ To create the `elasticsearch.keystore`, use the `create` command:
 
 [source,sh]
 ----------------------------------------------------------------
-bin/elasticsearch-keystore create
+bin/elasticsearch-keystore create -p
+----------------------------------------------------------------
+
+You will be prompted to enter the keystore password and the file `elasticsearch.keystore` will be created alongside `elasticsearch.yml` ,protected with the password you specified.
```
Comment (rjernst) [inline docs/reference/setup/secure-settings.asciidoc]: nit: spacing after comma, remove space before comma

**Review by rjernst — APPROVED** [review]
Looks good, I left some suggestions.

File: docs/reference/cluster/nodes-reload-secure-settings.asciidoc (hunk 1 above)
Comment (jkakavas) [reply docs/reference/cluster/nodes-reload-secure-settings.asciidoc]: I went ahead and removed the existing additional information. It felt more like describing internal implementation than adding something of value for the users/API consumers.

File: docs/reference/cluster/nodes-reload-secure-settings.asciidoc [hunk 8]
```diff
@@ -22,9 +16,41 @@ The first command reloads the keystore on each node. The seconds allows
 to selectively target `nodeId1` and `nodeId2`. The node selection options are
 detailed <<cluster-nodes,here>>.
 
-Note: It is an error if secure settings are inconsistent across the cluster
-nodes, yet this consistency is not enforced whatsoever. Hence, reloading specific
-nodes is not standard. It is only justifiable when retrying failed reload operations.
+NOTE: It is an error if secure settings are inconsistent across the cluster nodes, yet this consistency is not enforced
```
Comment (jrodewig) [inline docs/reference/cluster/nodes-reload-secure-settings.asciidoc]:
```suggestion
NOTE: {es} returns an error if secure settings are inconsistent across the cluster nodes, but this consistency is not enforced
```

File: docs/reference/cluster/nodes-reload-secure-settings.asciidoc [hunk 9]
```diff
@@ -22,9 +16,41 @@ The first command reloads the keystore on each node. The seconds allows
...
+whatsoever. Hence, reloading specific nodes is not standard.It is only justifiable when retrying failed reload operations.
```
Comment (jrodewig) [inline docs/reference/cluster/nodes-reload-secure-settings.asciidoc]:
```suggestion
whatsoever. Hence, reloading specific nodes is not standard. It is only justifiable when retrying failed reload operations.
```

File: docs/reference/setup/secure-settings.asciidoc [hunk 10]
```diff
@@ -32,16 +28,33 @@ To create the `elasticsearch.keystore`, use the `create` command:
...
+You will be prompted to enter the keystore password and the file `elasticsearch.keystore` will be created alongside `elasticsearch.yml`, protected with the password you specified.
+
+NOTE: If you don't specify the `-p` flag or if you enter an empty password, the elasticsearch keystore will be obfuscated but not password protected.
+
+[float]
+[[changing-keystore-password]]
+=== Changing the password of the keystore
+
+To change the password of the `elasticsearch.keystore`, use the `passwd` command.
+If the elasticsearch keystore is password protected, you will be prompted to enter the current password and then enter the new one
```
Comment (jrodewig) [inline docs/reference/setup/secure-settings.asciidoc]:
```suggestion
If the elasticsearch keystore is password protected, you will be prompted to enter the current password and then enter a new one.
```

File: docs/reference/setup/secure-settings.asciidoc [hunk 11]
```diff
@@ -32,16 +28,33 @@ To create the `elasticsearch.keystore`, use the `create` command:
...
+
+[source,sh]
+----------------------------------------------------------------
+bin/elasticsearch-keystore passwd
 ----------------------------------------------------------------
 
-The file `elasticsearch.keystore` will be created alongside `elasticsearch.yml`.
+You can use the `passwd` subcommand to set a password to a previously obfuscated only keystore, but also to remove the password from an encrypted keystore by setting it to an empty string.
```
Comment (jrodewig) [inline docs/reference/setup/secure-settings.asciidoc]:
```suggestion
You can use the `passwd` subcommand to set a password to a previously obfuscated-only keystore and  the password from an encrypted keystore by setting it to an empty string.
```

File: docs/reference/cluster/nodes-reload-secure-settings.asciidoc [hunk 12]
```diff
@@ -22,9 +16,41 @@ The first command reloads the keystore on each node. The seconds allows
...
+
+==== Reload Password Protected Secure Settings
+
+When the elasticsearch keystore is password protected and not simply obfuscated, the password for the keystore needs
```
Comment (jrodewig) [inline docs/reference/cluster/nodes-reload-secure-settings.asciidoc]:
```suggestion
When the {es} keystore is password protected and not simply obfuscated, the password for the keystore needs
```

File: docs/reference/cluster/nodes-reload-secure-settings.asciidoc [hunk 13]
```diff
@@ -22,9 +16,41 @@ The first command reloads the keystore on each node. The seconds allows
...
+[source,js]
+--------------------------------------------------
+POST _nodes/reload_secure_settings
+{
+  "reload_secure_settings": "s3cr3t" <1>
+}
+--------------------------------------------------
+// NOTCONSOLE
+
+<1> The common password that the elasticsearch keystore is encrypted with in every node of the cluster.
```
Comment (jrodewig) [inline docs/reference/cluster/nodes-reload-secure-settings.asciidoc]:
```suggestion
<1> The common password that the {es} keystore is encrypted with in every node of the cluster.
```

File: docs/reference/cluster/nodes-reload-secure-settings.asciidoc [hunk 14]
```diff
@@ -22,9 +16,41 @@ The first command reloads the keystore on each node. The seconds allows
...
+
+Alternatively the secure settings can be reloaded on a per node basis, locally accessing the API and passing the
+node-specific elasticsearch keystore password.
```
Comment (jrodewig) [inline docs/reference/cluster/nodes-reload-secure-settings.asciidoc]:
```suggestion
node-specific {es} keystore password.
```

File: docs/reference/cluster/nodes-reload-secure-settings.asciidoc [hunk 15]
```diff
@@ -22,9 +16,41 @@ The first command reloads the keystore on each node. The seconds allows
...
+[source,js]
+--------------------------------------------------
+POST _nodes/_local/reload_secure_settings
+{
+  "reload_secure_settings": "s3cr3t" <1>
+}
+--------------------------------------------------
+// NOTCONSOLE
+
+<1> The password that the elasticsearch keystore is encrypted with on the local node.
```
Comment (jrodewig) [inline docs/reference/cluster/nodes-reload-secure-settings.asciidoc]:
```suggestion
<1> The password that the {es} keystore is encrypted with on the local node.
```

File: docs/reference/setup/secure-settings.asciidoc [hunk 16]
```diff
@@ -32,16 +28,33 @@ To create the `elasticsearch.keystore`, use the `create` command:
 
 [source,sh]
 ----------------------------------------------------------------
-bin/elasticsearch-keystore create
+bin/elasticsearch-keystore create -p
+----------------------------------------------------------------
+
+You will be prompted to enter the keystore password and the file `elasticsearch.keystore` will be created alongside `elasticsearch.yml`, protected with the password you specified.
+
+NOTE: If you don't specify the `-p` flag or if you enter an empty password, the elasticsearch keystore will be obfuscated but not password protected.
```
Comment (jrodewig) [inline docs/reference/setup/secure-settings.asciidoc]:
```suggestion
NOTE: If you don't specify the `-p` flag or if you enter an empty password, the {es} keystore will be obfuscated but not password protected.
```

File: docs/reference/setup/secure-settings.asciidoc [hunk 17]
```diff
@@ -32,16 +28,33 @@ To create the `elasticsearch.keystore`, use the `create` command:
...
-The file `elasticsearch.keystore` will be created alongside `elasticsearch.yml`.
+You can use the `passwd` subcommand to set a password to a previously obfuscated only keystore, but also to remove the password from an encrypted keystore by setting it to an empty string.
 
 [float]
 [[list-settings]]
 === Listing settings in the keystore
 
-A list of the settings in the keystore is available with the `list` command:
+A list of the settings in the keystore is available with the `list` command.
+If the elasticsearch keystore is password protected, you will be prompted to enter the password:
```
Comment (jrodewig) [inline docs/reference/setup/secure-settings.asciidoc]:
```suggestion
If the {es} keystore is password protected, you will be prompted to enter the password:
```

File: docs/reference/setup/secure-settings.asciidoc [hunk 18]
```diff
@@ -52,8 +65,8 @@ bin/elasticsearch-keystore list
 [[add-string-to-keystore]]
 === Adding string settings
 
-Sensitive string settings, like authentication credentials for cloud
-plugins, can be added using the `add` command:
+Sensitive string settings, like authentication credentials for cloud plugins, can be added using the `add` command.
+If the elasticsearch keystore is password protected, you will be prompted to enter the password:
```
Comment (jrodewig) [inline docs/reference/setup/secure-settings.asciidoc]:
```suggestion
If the {es} keystore is password protected, you will be prompted to enter the password:
```

File: docs/reference/setup/secure-settings.asciidoc [hunk 19]
```diff
@@ -68,12 +81,22 @@ through stdin, use the `--stdin` flag:
...
+
+NOTE: The `-f` flag will also force the creation of an obfuscated-only keystore, if one doesn't already exist.
+
 [float]
 [[add-file-to-keystore]]
 === Adding file settings
 You can add sensitive files, like authentication key files for cloud plugins,
 using the `add-file` command. Be sure to include your file path as an argument
 after the setting name.
+If the elasticsearch keystore is password protected, you will be prompted to enter the password:
```
Comment (jrodewig) [inline docs/reference/setup/secure-settings.asciidoc]:
```suggestion
If the {es} keystore is password protected, you will be prompted to enter a password:
```

File: docs/reference/setup/secure-settings.asciidoc [hunk 20]
```diff
@@ -84,7 +107,8 @@ bin/elasticsearch-keystore add-file the.setting.name.to.set /path/example-file.j
 [[remove-settings]]
 === Removing settings
 
-To remove a setting from the keystore, use the `remove` command:
+To remove a setting from the keystore, use the `remove` command.
+If the elasticsearch keystore is password protected, you will be prompted to enter the password:
```
Comment (jrodewig) [inline docs/reference/setup/secure-settings.asciidoc]:
```suggestion
If the {es} keystore is password protected, you will be prompted to enter a password:
```

File: docs/reference/setup/secure-settings.asciidoc [hunk 21]
```diff
@@ -98,24 +122,29 @@ bin/elasticsearch-keystore remove the.setting.name.to.remove
...
+}
 ----
-// CONSOLE
-This API will decrypt and re-read the entire keystore, on every cluster node,
-but only the *reloadable* secure settings will be applied. Changes to other
-settings will not go into effect until the next restart. Once the call returns,
-the reload has been completed, meaning that all internal datastructures dependent
+// NOTCONSOLE
+
+<1> The password that the elasticsearch keystore is encrypted with.
```
Comment (jrodewig) [inline docs/reference/setup/secure-settings.asciidoc]:
```suggestion
<1> The password that the {es} keystore is encrypted with.
```

**Review by jrodewig — APPROVED** [review]
LGTM. Left a few minor nits and rewordings, but nothing blocking. Thanks @jkakavas.

File: docs/reference/cluster/nodes-reload-secure-settings.asciidoc (hunk 8 above)
Comment (jkakavas) [reply docs/reference/cluster/nodes-reload-secure-settings.asciidoc]: We don't return an error ( this is what the "but this consistency is not enforced" is meant to explain ). I see how this is not very clear, I'm open to suggestions to rephrase this

Comment (jrodewig) [reply docs/reference/cluster/nodes-reload-secure-settings.asciidoc]: Thanks for the clarification. Here's a suggested rephrasing. 

```suggestion
NOTE: {es} requires consistent secure settings across the cluster nodes, but this consistency is not enforced.
```

Let me know if this is accurate. Also note that this removes the "whatsover." on the next line.
